boxoffice/models/invoice.py

```diff
@@ -72,6 +73,20 @@ def __init__(self, *args, **kwargs):
         self.invoice_no = gen_invoice_no(organization, country_code, self.invoiced_at)
         super(Invoice, self).__init__(*args, **kwargs)
 
+    @property
+    def is_final(self):
+        return self.status == INVOICE_STATUS.FINAL
+
+@event.listens_for(Invoice, 'before_update')
+def validate_immutable_finalized_invoice(mapper, connection, target):
+    """
+    Raises ValueError if a finalized invoice is being modified. Allows the status to be toggled.
+    """
+    # get_history helps check if the invoice was previously in draft status.
+    # This check is necessary to allow edits to an invoice which may be transitioning from draft to final
+    if target.is_final and get_history(target, 'status').deleted.count(INVOICE_STATUS.DRAFT) == 0:
```

Comment:
Looks like you need `if inspect(target).attrs.status.loaded_value == INVOICE_STATUS.FINAL:`

Reply:
> If the value has not been loaded, or is otherwise not present in the object’s dictionary, returns NO_VALUE.

So `if inspect(target).attrs.status.loaded_value in (INVOICE_STATUS.FINAL, symbol('NO_VALUE')):`

This will prevent any changes to the `Invoice` if the `status` field was not loaded from the database, which I guess is okay.